### PR TITLE
Reinstated the service metrics

### LINKED ISSUE #461: failed to create or get service for metrics
Currently, the Jaeger Operator fails to start in OpenShift due to the following message:

```
time="2019-06-05T12:29:54Z" level=fatal msg="failed to create or get service for metrics: services \"jaeger-operator\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"
```

This log message is recorded with `fatal` level, making the operator to simply exit upon this failure. Operators created with newer versions of the SDK have this set to `info` instead. This of course doesn't fix the underlying problem, but I think we should not crash the operator because of this failure.

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #461 by adding the required rule in the role.yaml file.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: pkg/cmd/start/main.go [hunk 1]
```diff
@@ -133,6 +140,51 @@ func start(cmd *cobra.Command, args []string) {
 		log.Fatal(err)
 	}
 
+	// Create Service object to expose the metrics port.
+	operatorService, err := metrics.ExposeMetricsPort(ctx, viper.GetInt32("metrics-port"))
+	if err != nil {
+		log.Fatal(err)
+	} else if operatorService != nil {
+		if err := setOwnerReference(operatorService); err != nil {
```
Comment (LiliC) [inline pkg/cmd/start/main.go]: This is not needed we already do this in operator-sdk `metrics.ExposeMetricsPort` https://github.com/operator-framework/operator-sdk/blob/v0.8.1/pkg/metrics/metrics.go#L109

Comment (jpkrohling) [reply pkg/cmd/start/main.go]: @objectiser: I think you added this originally -- do you remember what problem it was solving?

Comment (objectiser) [reply pkg/cmd/start/main.go]: @jpkrohling The metrics service was not being removed.

Comment (jpkrohling) [reply pkg/cmd/start/main.go]: To test this, is it sufficient to not set the owner reference, create the operator, delete the operator, and check which services are available in the namespace?

Comment (objectiser) [reply pkg/cmd/start/main.go]: That should be adequate.

Comment (jpkrohling) [reply pkg/cmd/start/main.go]: Alright, I just confirmed it does remove the service when the operator is undeployed. I removed the extra code.

```console
$ kubectl get services
No resources found.
$ kubectl apply -f deploy/operator.yaml 
deployment.apps/jaeger-operator created
$ kubectl get pods
NAME                               READY     STATUS    RESTARTS   AGE
jaeger-operator-774d9c9f66-zxhz7   1/1       Running   0          16s
$ kubectl get services
NAME              TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
jaeger-operator   ClusterIP   172.30.103.38   <none>        8383/TCP   14s
$ kubectl delete deployment jaeger-operator
deployment.extensions "jaeger-operator" deleted
$ kubectl get services
No resources found.
```

**Pull request merged**